docs/react/README.md

```diff
@@ -1,72 +1,186 @@
 # @hatchifyjs/react
 
-- [What is @hatchifyjs/react](#what-is-hatchifyjsreact)
-  - [TypeScript](#typescript)
-  - [Types](#types)
+@hatchifyjs/react is a schema-driven library of components and helper methods for a Hatchify + React frontend. By providing the schemas (AKA models) of your backend resources, @hatchifyjs/react returns a set of components and helper functionality that you can use across your React app.
+
+`@hatchifyjs/react` is a TypeScript supported [NPM package](https://www.npmjs.com/package/@hatchifyjs/react) that takes your [JSON:API rest client](#createjsonapiclient) and[Schema(s)](../schema/README.md) to produce custom:
+
+- components
+- helper functions
+- generated [types](#types)
+
+The following uses `hatchifyReact`to create a `hatchedReact` app with the defined `Todo` and `User` schemas.
+
+```tsx
+import { hatchifyReact, HatchifyProvider, createJsonapiClient } from "@hatchifyjs/react"
+
+// Define your Schema(s)
+export const Schemas = {
+  Todo: {
+    name: "Todo",
+    attributes: {
+      name: string(),
+      status: enumerate({ values: ["Pending", "Failed", "Completed"] }),
+    },
+    relationships: {
+      user: belongsTo("User"),
+    },
+  },
+  User: {
+    name: "User",
+    attributes: {
+      name: string(),
+    },
+    relationships: {
+      todos: hasMany(),
+    },
+  },
+} satisfies Record<string, PartialSchema>
+
+// Create your Hatched React App instance
+const hatchedReact = hatchifyReact(createJsonapiClient("/api", Schemas))
+
+// Define variables for your Hatchify schema generated components
+const TodoDataGrid = hatchedReact.components.Todo.DataGrid
+
+// Render your React Functional Component
+const HatchedComponent = () => {
+  return (
+    <HatchifyProvider>
+      <button onClick={onActionClick} style={{ margin: 10 }}>
+        action
+      </button>
+      <TodoDataGrid defaultSort={{ direction: "asc", sortBy: "id" }}></TodoDataGrid>
+    </HatchifyProvider>
+  )
+}
+
+export default App
+```
+
+[!IMPORTANT]
+The `HatchifyProvider` wraps all Hatchify components to manage their internal states. See [HatchifyProvider](#hatchifyprovider) for more information.
+
+- [Exports](#exports)
   - [createJsonapiClient](#createjsonapiclient)
   - [HatchifyProvider](#hatchifyprovider)
   - [hatchifyReact](#hatchifyreact)
-    - [Everything](#everything)
-    - [components](#components)
-    - [model](#model)
-    - [state](#state)
-      - [useDataGridState](#usedatagridstate)
-  - [MUI Components](#mui-components)
+  - [Types](#types)
+    - [DataGridState](#usedatagridstate)
+    - [CreateType](#usedatagridstate)
+    - [UpdateType](#usedatagridstate)
+    - [HatchifyApp](#usedatagridstate)
+    - [RecordType](#usedatagridstate)
+- [`hatchedReact`](#hatchedreact)
+  - [`hatchedReact.Everything`](#everything)
```

Comment:
since these are nested under `hatchedReact` i think you can remove the prefix, so we just have:

- hatchedReact
  - Everything
  - components
  ...